examples/interactive_remote_hyperdrive_example.py

```diff
@@ -33,63 +25,142 @@
+)
+# We can initialize an agent with a custom policy - more on that later
+agent1 = chain.init_agent(
+    private_key=private_key_1,
+    name="agent1",
     policy=PolicyZoo.random,
-    # The configuration for the underlying policy
-    policy_config=PolicyZoo.random.Config(rng_seed=123),
+    policy_config=PolicyZoo.random.Config(),
 )
 
 # %%
-# We expose this function for testing purposes, but the underlying function calls `mint` and `anvil_set_balance`,
-# which are likely to fail on any non-test network.
-agent0.add_funds(base=FixedPoint(100000), eth=FixedPoint(100))
+# Connect to a hyperdrive pool
+
+# Define a specific pool address
+# hyperdrive_address = "0x0000000000000000000000000000000000000000"
+
+# Alternatively, look up the list of registered hyperdrive pools
+# This is the registry address deployed on sepolia.
+# registry_address = "0xba5156E697d39a03EDA824C19f375383F6b759EA"
+#
+# hyperdrive_address = Hyperdrive.get_hyperdrive_addresses_from_registry(chain, registry_address)["sdai_14_day"]
 
-# Set max approval
-agent0.set_max_approval()
+# We'll use the previously deployed pool for this exapmle
+
+hyperdrive_config = Hyperdrive.Config()
+hyperdrive_pool = Hyperdrive(chain, hyperdrive_address, hyperdrive_config)
 
 # %%
+# We expose this function for testing purposes, but the underlying function
+# calls `mint` and `anvil_set_balance`, which are likely to fail on any non-test
+# network. In practice, it's up to the user to ensure the wallet has sufficient funds.
+agent0.add_funds(base=FixedPoint(100000), eth=FixedPoint(100), pool=hyperdrive_pool)
+agent1.add_funds(base=FixedPoint(100000), eth=FixedPoint(100), pool=hyperdrive_pool)
+
+
+# %% [markdown]
+#####################
+# Executing Trades
+#####################
+
+# We set agent1's active pool to avoid passing in pool for functions.
+agent1.set_active(pool=hyperdrive_pool)
 
+# Set max approval for the agent on a specific pool.
+# TODO this is overly permissive, we may want to expose setting approval for a specific
+# amount, or allow a parameter in trade calls to approve for the preview amount
+# (plus slippage) before executing a trade.
+agent0.set_max_approval(pool=hyperdrive_pool)
+agent1.set_max_approval()
+
+# %%
 # Make trades
+
 # Return values here mirror the various events emitted from these contract calls
-# These functions are blocking, but relatively easy to expose async versions of the
-# trades below
-open_long_event = agent0.open_long(base=FixedPoint(11111))
-close_long_event = agent0.close_long(maturity_time=open_long_event.maturity_time, bonds=open_long_event.bond_amount)
+# TODO These functions are blocking, expose async versions of the trades below
+open_long_event = agent0.open_long(base=FixedPoint(11111), pool=hyperdrive_pool)
+close_long_event = agent0.close_long(
+    maturity_time=open_long_event.maturity_time,
+    bonds=open_long_event.bond_amount,
+    pool=hyperdrive_pool,
+)
+
+open_short_event = agent1.open_short(bonds=FixedPoint(33333))
```

Comment:
We probably don't need lines 104-110, since it's redundant with the local example? I'm not sure how much we should repeat ourselves. Your call.

Reply:
These analysis functions are a subset of the supported ones in local, so I wanted to give examples of the allowed queries in the remote case